### PR TITLE
add csp

### PR DESCRIPTION
**Pull request by timkpaine**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Note: this library is undergoing a name-change so the import name does not match the library name (as of this moment) https://github.com/timkpaine/sp2/issues/11

### CODE REVIEW

**Comment by timkpaine** [comment]
@conda-forge/help-python all set

**Comment by timkpaine** [comment]
@conda-forge/help-python @conda-forge/help-python-c all set, thanks! I will enable Mac build once the feedstock is rendered. Windows is still WIP upstream, so leaving that disabled for now.

**Comment by timkpaine** [comment]
ref macOS: https://github.com/stream-processing/sp2/issues/36

File: recipes/sp2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - make  # [not win]
+  host:
+    - python
+    - bison
```
Comment (xhochy) [inline recipes/sp2/meta.yaml:35]: `bison` should normally be in build as it is used as a tool

File: recipes/sp2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    - numpy
+    - pip
+    - pyarrow =15
+    - rapidjson
+    - ruamel.yaml
+    - scikit-build
+    - setuptools >=69
+    - tar
+    - unzip
+    - zip
```
Comment (xhochy) [inline recipes/sp2/meta.yaml:56]: They also should be in build
```suggestion
```

File: recipes/sp2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    - lz4-c
+    - packaging
+    - pandas <2.2
+    - pyarrow =15
+    - psutil
+    - pytz
+    - rapidjson
+    - ruamel.yaml
+    - sqlalchemy
+    - zlib
```
Comment (xhochy) [inline recipes/sp2/meta.yaml:76]: Most of these come in automatically via `run_exports`. Please then also let `run_exports` do its thing as some of them only install a subset of the development package at runtime.
```suggestion
    - packaging
    - pandas <2.2
    - pyarrow =15
    - psutil
    - pytz
    - rapidjson
    - ruamel.yaml
    - sqlalchemy
```

File: recipes/sp2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+test:
```
Comment (xhochy) [inline recipes/sp2/meta.yaml:78]:
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**